deployDataFactoryTask/task.json

```diff
@@ -69,7 +69,7 @@
             "type": "string",
             "label": "Target Region",
             "helpMarkDown": "Azure Region of target Data Factory.\nIt is used only in order to create new instance of ADF.",
-            "required": true
+            "required": false
```

Comment:
Why this has been changed?

Reply:
Apologies, I forgot about this one. I felt it was a good change because the parameter isn't mandatory in all cases. If I choose not to allow creation of new ADF instances during deployment, and if this parameter is mandatory, then I have to fill in the field with some dummy value, such as "n/a". Therefore, I think it's better to leave it optional and have the task report an error if the field is blank but the Factory needs to be created.

Reply:
I would prefer to leave it as is, because it may confuse people. Unfortunately, there is no option to apply an expression here and change the "required" based on the value of another field.
Also, it should be covered by separate issue & PR, if at all.

Reply:
No worries, I've removed that change, also have rebased the change off your current master.